src/core/Link/RouterLink/RouterLink.tsx

```diff
@@ -0,0 +1,101 @@
+// import React, { Component, ReactNode } from 'react';
+import React, { ReactNode } from 'react';
+import { default as styled } from 'styled-components';
+import { RouterLinkStyles } from './RouterLink.baseStyles';
+import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../theme';
+import { baseClassName } from '../BaseLink/BaseLink';
+import classnames from 'classnames';
+import { HtmlA } from '../../../reset';
+
+//
+// Source: https://www.benmvp.com/blog/polymorphic-react-components-typescript/
```

Comment:
Everything in this file until line 72 is copied directly from https://github.com/vrk-kpa/suomifi-ui-components/tree/fix/as-prop branch (work done by @ketsappi)